pyaedt/edb_core/edb_data/hfss_extent_info.py

```diff
@@ -0,0 +1,328 @@
+            (value, self.air_box_negative_vertical_extent_enabled)
+        )
+        self._update_hfss_extent_info(info)
+
+    @property
+    def base_polygon(self):
+        """Get base polygon
+
+        Returns
+        -------
+        :class:`pyaedt.edb_core.edb_data.primitives_data.EDBPrimitive`
+        """
+        return EDBPrimitives(self._edb_hfss_extent_info.BasePolygon, self._pedb)
+
+    @base_polygon.setter
+    def base_polygon(self, value):
+        info = self._edb_hfss_extent_info
+        info.BasePolygon = value.primitive_object
+        self._update_hfss_extent_info(info)
+
+    @property
+    def dielectric_base_polygon(self):
+        """Get dielectric base polygon.
+
+        Returns
+        -------
+        :class:`pyaedt.edb_core.edb_data.primitives_data.EDBPrimitive`
+        """
+        return EDBPrimitives(self._edb_hfss_extent_info.DielectricBasePolygon, self._pedb)
+
+    @dielectric_base_polygon.setter
+    def dielectric_base_polygon(self, value):
+        info = self._edb_hfss_extent_info
+        info.DielectricBasePolygon = value.primitive_object
+        self._update_hfss_extent_info(info)
+
+    @property
+    def dielectric_extent_size_enabled(self):
+        """Enable dielectric extent size."""
+        return self._edb_hfss_extent_info.DielectricExtentSize.Item2
+
+    @dielectric_extent_size_enabled.setter
+    def dielectric_extent_size_enabled(self, value):
+        info = self._edb_hfss_extent_info
+        info.DielectricExtentSize = convert_pytuple_to_nettuple((self.dielectric_extent_size, value))
+        self._update_hfss_extent_info(info)
+
+    @property
+    def dielectric_extent_size(self):
+        """Get dielectric extent size."""
+        return self._edb_hfss_extent_info.DielectricExtentSize.Item1
+
+    @dielectric_extent_size.setter
+    def dielectric_extent_size(self, value):
+        info = self._edb_hfss_extent_info
+        info.DielectricExtentSize = convert_pytuple_to_nettuple((value, self.dielectric_extent_size_enabled))
+        self._update_hfss_extent_info(info)
+
+    @property
+    def dielectric_extent_type(self):
+        """Get dielectric extent type."""
+        return self._edb_hfss_extent_info.DielectricExtentType.ToString()
+
+    @dielectric_extent_type.setter
+    def dielectric_extent_type(self, value):
+        info = self._edb_hfss_extent_info
+        info.DielectricExtentType = self._hfss_extent_info_type[value]
+        self._update_hfss_extent_info(info)
+
+    @property
+    def extent_type(self):
+        """Get extent type."""
+        return self._edb_hfss_extent_info.ExtentType.ToString()
+
+    @extent_type.setter
+    def extent_type(self, value):
+        info = self._edb_hfss_extent_info
+        info.ExtentType = self._hfss_extent_info_type[value]
+        self._update_hfss_extent_info(info)
+
+    @property
+    def honor_user_dielectric(self):
+        """Get honor user dielectric."""
```

Comment:
```suggestion
        """Honor user dielectric."""
```
This seems like it needs to have more description?